feat(Response): add response.remoteAddress method

szdc opened the pull request:
This patch exposes a remoteAddress property which is a formatted string containing the IP address and port of the remote server

Closes #2171 

I wasn't sure if I should expose the remote address and port separately (rather than a formatted version), so please let me know if that's a better idea.

szdc commented:
Thoughts on how I tackle the IPv4 loopback address in the Travis build?

aslushnikov commented on lib/NetworkManager.js:
```diff
@@ -188,7 +188,7 @@ class NetworkManager extends EventEmitter {
    */
   _handleRequestRedirect(request, redirectStatus, redirectHeaders, fromDiskCache, fromServiceWorker, securityDetails) {
-    const response = new Response(this._client, request, redirectStatus, redirectHeaders, fromDiskCache, fromServiceWorker, securityDetails);
+    const response = new Response(this._client, request, redirectStatus, redirectHeaders, fromDiskCache, fromServiceWorker, securityDetails, null, null);
```
let's plumb the remoteIDAddress and remotePort to the redirect responses (also let's cover this with a test)

szdc replied:
Good point - I'll update with a test later today.

aslushnikov commented on examples/search.js:
```diff
@@ -27,8 +27,6 @@ const puppeteer = require('puppeteer');
   const browser = await puppeteer.launch();
   const page = await browser.newPage();
 
-  await page.goto('https://developers.google.com/web/');
```
why this change?

szdc replied:
Accidental; sorry.

aslushnikov commented on docs/api.md:
```diff
@@ -2513,6 +2514,9 @@ This method will throw if the response body is not parsable via `JSON.parse`.
 Contains a boolean stating whether the response was successful (status in the range 200-299) or not.
 
+#### response.remoteAddress()
+- returns: <[string]> The formatted address and port of the remote server.
```
I'd return IP address and port separately: it's a little error-prone to parse out port number with IPv6 addresses.

For example, this method can return an object `{ip, port}`

aslushnikov commented on test/network.spec.js:
```diff
@@ -56,6 +56,7 @@ module.exports.addTests = function({testRunner, expect}) {
       expect(responses[0].fromCache()).toBe(false);
       expect(responses[0].fromServiceWorker()).toBe(false);
       expect(responses[0].request()).toBeTruthy();
+      expect(responses[0].remoteAddress()).toBe(`[::1]:${server.PORT}`);
```
I'd try navigating to IP explicitly in line 51 to make this less flaky across environments (different setups might resolve
`localhost` differently)

```js
await page.goto(`http://[::1]:${server.PORT}/empty.html`);
```

szdc replied:
We'll still have trouble on IPv4-only machines as they won't be able to handle `[::1]`.  If we use `127.0.0.1` instead, Chrome still returns `[::1]` on IPv6-capable machines.

szdc committed:
feat(Response): add response.remoteAddress method

This patch exposes a remoteAddress property which is a formatted string containing the IP address and port of the remote server

Closes #2171

aslushnikov commented:
Close in favor of #3192.
